core/domain/user_services.py

```diff
@@ -713,8 +729,22 @@ def get_usernames(user_ids):
         If a user_id does not exist, the corresponding entry in the
         returned list is None.
     """
+    usernames = [''] * len(user_ids)
+    for index, user_id in enumerate(user_ids):
+        if (user_id == feconf.SYSTEM_COMMITTER_ID or
+                user_id == feconf.MIGRATION_BOT_USER_ID):
+            usernames[index] = get_system_username(user_id)
+
     users_settings = get_users_settings(user_ids)
-    return [us.username if us else None for us in users_settings]
+
+    for index, user_settings in enumerate(users_settings):
+        if usernames[index] == '':
```

Comment:
That's icky. I don't think you should be calling get_users_settings with anything that is ''. I think you'll need to pull out a list of non-system user ids and their original indexes, do the get_multi, and then put them back in.

Also, this needs backend tests.

Reply:
Agree :P, Done.